### PR TITLE
Make count queries publicly available for use

### PR DESCRIPTION
**Pull request by milaGGL**
This PR publicly exports the classes, methods, and functions that provide "count" queries, making them available for anyone to use. See https://github.com/firebase/firebase-js-sdk/pull/6597 for the implementation of COUNT queries.

Withholding this PR till the count query is ready to be released.

### CODE REVIEW

**Review by dconeybe — CHANGES_REQUESTED** [review]
Please add a change log entry (i.e. a changeset) with a message something like

"Expose getCountFromServer() to enable Firestore count-only queries"

Otherwise LGTM. Let's just get someone from the JS team to take another look.

File: packages/firestore/test/integration/api/aggregation.test.ts
Comment (milaGGL) [reply packages/firestore/test/integration/api/aggregation.test.ts:24]:
```
import {
  collection,
  collectionGroup,
  disableNetwork,
  doc,
  DocumentData,
  getCountFromServer,
  query,
  QueryDocumentSnapshot,
  terminate,
  where,
  writeBatch
} from '../util/firebase_export';
```
Used automatic sorting helper to sort the imports, not sure which one is out of order. 🤔

Comment (dconeybe) [reply packages/firestore/test/integration/api/aggregation.test.ts:24]: I guess as long as the linter isn't complaining, don't worry about it.

**Comment by dconeybe** [comment]
nit: Spelling error in this PR's title: "qury"

**Comment by ehsannas** [comment]
@milaGGL Please rebase this PR against `master`

File: .changeset/wicked-tomatoes-grow.md
Comment (dconeybe) [inline .changeset/wicked-tomatoes-grow.md:5]: Please use this wording, to be consistent with the iOS and Android SDKs: 

```
Added `getCountFromServer()` (`getCount()` in the Lite SDK), which fetches the number of documents in the result set without actually downloading the documents.
```

File: packages/firestore/src/api/aggregate.ts
Comment (dconeybe) [inline packages/firestore/src/api/aggregate.ts:25]: Is it possible to remove these exports and instead have api.ts directly export from the lite sdk? I'd just like to avoid this indirection if at all possible.

**Review by dconeybe — CHANGES_REQUESTED** [review]

File: packages/firestore/src/core/count_query_runner.ts
Comment (dconeybe) [inline packages/firestore/src/core/count_query_runner.ts:28]: Can you drop the `<T>` type parameter in `CountQueryRunner<T>`? It appears to be unused (according to lint)>

**Review by dconeybe — CHANGES_REQUESTED** [review]
One tiny change. Also, please address the problems reported by lint.

**Comment by dconeybe** [comment]
Here's another problem: When I ran `yarn build` it produced this error:

```
Error: Circular dependency: src/lite-api/aggregate.ts -> src/core/count_query_runner.ts -> src/lite-api/aggregate.ts
```

This need to be fixed too.

**Review by dconeybe — APPROVED** [review]
LGTM!

**Comment by dconeybe** [comment]
I've added the do-not-merge label to make sure we don't merge until we get the green light to release count.

File: .changeset/wicked-tomatoes-grow.md
Comment (hsubox76) [inline .changeset/wicked-tomatoes-grow.md:2]: Can you also add `'firebase': minor`? Unfortunately changesets can't autodetect that a minor or major bump in a dependency ought to bump the main `firebase` package by the same increment so we have to add it manually.

Comment (dconeybe) [reply .changeset/wicked-tomatoes-grow.md:2]: Done

**Review by hsubox76 — APPROVED** [review]

File: packages/firestore/src/api/aggregate.ts
Comment (egilmorez) [inline packages/firestore/src/api/aggregate.ts:32]: This comma can be omitted.

File: packages/firestore/src/api/aggregate.ts
Comment (egilmorez) [inline packages/firestore/src/api/aggregate.ts:36]: We should omit "even"

File: packages/firestore/src/api/aggregate.ts
Comment (egilmorez) [inline packages/firestore/src/api/aggregate.ts:37]: Suggest "count the documents in cases where the result set would be prohibitively large to
 * download entirely (thousands of documents)."
 
 We avoid "e.g." and "i.e."

File: packages/firestore/src/api/aggregate.ts
Comment (egilmorez) [inline packages/firestore/src/api/aggregate.ts:44]: This colon should be a period or possibly a semicolon.

File: packages/firestore/src/api/aggregate.ts
Comment (egilmorez) [inline packages/firestore/src/api/aggregate.ts:47]: Suggest "The query whose result set size is calculated."

File: packages/firestore/src/api/aggregate.ts
Comment (egilmorez) [inline packages/firestore/src/api/aggregate.ts:32]: I think all of this comment is copied in FS lite, so all feedback applies there too, thanks!

File: packages/firestore/src/core/count_query_runner.ts
Comment (egilmorez) [inline packages/firestore/src/core/count_query_runner.ts:28]: Should this just begin with "Encapsulates . . ."  ?

If we keep this, backtick as a literal.

**Review by egilmorez — COMMENTED** [review]
Here are some things that would be great to get into a follow-on PR, thanks!

**Comment by dconeybe** [comment]
> Here are some things that would be great to get into a follow-on PR, thanks!

Thanks for the suggestions! I was the one who wrote these doc comments so I'll make the follow-up changes. I'll be tracking this work in b/250612059.

**Comment by egilmorez** [comment]
Sounds good, thanks Denver!

File: packages/firestore/src/api/aggregate.ts
Comment (dconeybe) [reply packages/firestore/src/api/aggregate.ts:32]: PR created: https://github.com/firebase/firebase-js-sdk/pull/7933

File: packages/firestore/src/api/aggregate.ts
Comment (dconeybe) [reply packages/firestore/src/api/aggregate.ts:37]: PR created: https://github.com/firebase/firebase-js-sdk/pull/7933

File: packages/firestore/src/api/aggregate.ts
Comment (dconeybe) [reply packages/firestore/src/api/aggregate.ts:47]: PR created: https://github.com/firebase/firebase-js-sdk/pull/7933

File: packages/firestore/src/core/count_query_runner.ts
Comment (dconeybe) [reply packages/firestore/src/core/count_query_runner.ts:28]: PR created: https://github.com/firebase/firebase-js-sdk/pull/7933

File: packages/firestore/src/api/aggregate.ts
Comment (dconeybe) [reply packages/firestore/src/api/aggregate.ts:32]: PR created: https://github.com/firebase/firebase-js-sdk/pull/7933

File: packages/firestore/src/api/aggregate.ts
Comment (dconeybe) [reply packages/firestore/src/api/aggregate.ts:44]: PR created: https://github.com/firebase/firebase-js-sdk/pull/7933

File: packages/firestore/src/api/aggregate.ts
Comment (dconeybe) [reply packages/firestore/src/api/aggregate.ts:36]: PR created: https://github.com/firebase/firebase-js-sdk/pull/7933

**Pull request merged**
